runtime/onert/core/src/compiler/TensorRegistries.h

```diff
@@ -52,6 +55,15 @@ class TensorRegistries
       {
         _tensor_regs.insert(tensor_reg);
       }
+
+      if (e.first->config()->supportTraining())
```

Comment:
I suggest to define another TensorRegistries constructor for TrainableExecutor.

Reply:
`BackendContexts` may contain backends such as the`builtin` backend that do not support training. I don't think it can be solved with the ways you suggested.